### PR TITLE
NCBC-3678: Docs samples for SDK 3.5 features

### PR DESCRIPTION
**Pull request by RiPont**
* Scoped search examples
* Vector search examples
* refactored from inline examples to CSPROJ example that compiles.

### CODE REVIEW

File: modules/howtos/examples/Couchbase.Examples.SearchV2/SearchV2Examples/Program.cs [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+    .MinimumLevel.Verbose()
+    .WriteTo.Console()
+    .CreateLogger();
+
+var clusterOptions = new ClusterOptions()
+{
+    ConnectionString = "couchbase://localhost",
+    UserName = "Administrator",
+    Password = "password",
+    EnableDnsSrvResolution = false,
```
Comment (emilienbev) [inline modules/howtos/examples/Couchbase.Examples.SearchV2/SearchV2Examples/Program.cs:23]: NIT: tiny extra "," here

**Review by jeffrymorris — CHANGES_REQUESTED** [review]
Hi Richard -

I think you are missing KV Range Scan  and Read from Replica docs. via https://issues.couchbase.com/browse/CBD-5804

File: modules/howtos/examples/Couchbase.Examples.SearchV2/SearchV2Examples/Program.cs (hunk 1 above)
Comment (RiPont) [reply modules/howtos/examples/Couchbase.Examples.SearchV2/SearchV2Examples/Program.cs:23]: That's idiomatic modern C#.  :)  Compiles just fine.

**Review by RichardSmedley — APPROVED** [review]
Thanks @RiPont 
LGTM

**Pull request merged**